tests/unit/src/main/scala/tests/BaseRangesSuite.scala

```diff
@@ -0,0 +1,63 @@
+package tests
+
+import munit.Location
+import scala.concurrent.Future
+
+abstract class BaseRangesSuite(name: String) extends BaseLspSuite(name) {
+
+  def assertCheck(
+      filename: String,
+      edit: String,
+      expected: Map[String, String],
+      base: Map[String, String]
+  ): Future[Unit]
+
+  def check(name: String, input: String)(implicit loc: Location): Unit = {
+    val files = FileLayout.mapFromString(input)
+    val (filename, edit) = files
+      .find(_._2.contains("@@"))
+      .map {
+        case (fileName, code) =>
+          (fileName, code.replaceAll("(<<|>>)", ""))
+      }
+      .getOrElse {
+        throw new IllegalArgumentException(
+          "No `@@` was defined that specifies cursor position"
+        )
+      }
+    val expected = files.map {
```

Comment:
So I originally had written a test that sorted the `Location`s that all came back and asserted that they were in the same order. However, I realized (I hope correctly), that this will actually implicitly test the same thing. If for some reason the references returned are out of order, this expected mapping with the string representation of what we gave the test won't match later on when we assert that the references returned match this. Let me know if this is indeed a good enough test for the actual ordering.

Reply:
I don't think it will work that way though. We just replace all the ranges regardless of their sorting. But you can just add an assert . I will point to where to add it.